+nosnoc/+discrete_time_problem/Stewart.m

```diff
@@ -413,38 +467,111 @@ function generate_direct_transcription_constraints(obj)
                     model.x_ref_end_val,...
                     p_global);
             end
+
+            %  -- Constraint for the terminal numerical and physical time (if no equidistant grids are required) --
+            % If the control grid is not equidistant, the constraint on sum of h happen only at the end.
+            % The constraints are split to those which are related to numerical and physical time, to make it easier to read.
+            if opts.time_freezing
```

Comment:
the code here is identical to the one in Heaviside.. not sure does it make sense to extract it into a function (or at least copy all the explanation comments (cf. comment in Heaviside on explaining every branching)

Reply:
Perhaps it may make sense to extract this though I am unsure. I think we can leave it as is for now and if having two copies is inconvenient we can extract then.